src/ring_theory/localization/basic.lean

```diff
@@ -806,6 +806,24 @@ instance : comm_ring (localization M) :=
   right_distrib  := λ m n k, localization.induction_on₃ m n k (by tac),
    ..localization.comm_monoid M }
 
+lemma sub_mk (a c) (b d) : (mk a b : localization M) - mk c d = mk (d * a - b * c) (b * d) :=
+calc  mk a b - mk c d
+    = mk a b + (- mk c d) : rfl
+... = mk a b + (mk (-c) d) : by rw neg_mk
+... = mk (b * (-c) + d * a) (b * d) : add_mk _ _ _ _
+... = mk (d * a - b * c) (b * d) : by congr'; ring
+
+lemma coe_int (m : ℤ) : (m : localization M) = mk m 1 :=
+int.induction_on m (mk_zero _).symm (λ n ih, begin
+  push_cast at ⊢ ih,
+  erw [ih, ← mk_one, add_mk, one_mul, one_mul, one_mul, add_comm],
+end) $ λ n ih, begin
+  push_cast at ⊢ ih,
+  erw [ih, ← mk_one, sub_mk, one_mul, one_mul, one_mul]
+end
+
+lemma coe_nat (m : ℕ) : (m : localization M) = mk m 1 := coe_int m
+
```

Comment:
These ended up not being used; should I keep them

Reply:
I think there should be a shoter proof of `coe_int` using some statement about `mk m 1` being a ring_hom.

I also think that perhaps `mk_nat_cast : mk m 1 = (m : localization M)` might be a better statement, to match the direction of `mk_zero`.

Otherwise I think these lemmas are definitely worth having - but it would make sense to do them in a separate PR if you don't use them here.

Reply:
I have found a much shorter proof.
I will open another PR.

Reply:
see #12861